plugins/src/options.rs

```diff
@@ -63,95 +84,188 @@ impl Value {
             _ => None,
         }
     }
+}
 
-    /// Return `true` if the option is not None and `false` otherwise.
-    pub fn is_some(&self) -> bool {
-        match self {
-            Value::String(_) => false,
-            Value::Integer(_) => false,
-            Value::Boolean(_) => false,
-            Value::OptString => true,
-            Value::OptInteger => true,
-            Value::OptBoolean => true,
+#[derive(Clone, Debug)]
+pub struct ConfigOption<V: OptionType> {
+    name: String,
+    default: Option<V>,
+    value_type: ValueType,
+    description: String,
+}
+
+impl ConfigOption<String> {
+    pub fn build(&self) -> UntypedConfigOption {
+        UntypedConfigOption {
+            name: self.name.clone(),
+            value_type: self.value_type.clone(),
+            default: self.default.as_ref().map(|s| Value::String(s.clone())),
+            description: self.description.clone(),
+            value: None,
+        }
+    }
+
+    pub fn new_str_with_default<S1: AsRef<str>, S2: AsRef<str>, S3: AsRef<str>>(
+        name: S1,
+        default: S2,
+        description: S3,
+    ) -> Self {
+        Self {
+            name: name.as_ref().to_string(),
+            default: Some(default.as_ref().to_string()),
+            value_type: ValueType::String,
+            description: description.as_ref().to_string(),
+        }
+    }
+}
+
+impl ConfigOption<i64> {
+    pub fn build(&self) -> UntypedConfigOption {
+        UntypedConfigOption {
+            name: self.name.clone(),
+            value_type: self.value_type.clone(),
+            default: self.default.map(|i| Value::Integer(i)),
+            description: self.description.clone(),
+            value: None,
+        }
```

Comment:
Interesting, would you say this is an example of type-erasure? I've been looking for a good example to see why that is useful :-)